pkg/platform/authentication/auth.go

```diff
@@ -113,9 +116,40 @@ func (s *Auth) Close() error {
 	return nil
 }
 
+func (s *Auth) storeAuthenticatedDevice(deviceCode strfmt.UUID, response *oauth.AuthDeviceGetOK) error {
+	defer s.updateRollbarPerson()
+	s.user = response.Payload.AccessToken.User
+	s.bearerToken = response.Payload.AccessToken.Token
+	clientAuth := httptransport.BearerToken(s.bearerToken)
+	s.clientAuth = &clientAuth
+	err := s.cfg.Set(deviceCodeConfigKey, deviceCode)
+	if err != nil {
+		return errs.Wrap(err, "Could not set deviceCode credentials in config")
+	}
+	return err
+}
+
 // Authenticated checks whether we are currently authenticated
 func (s *Auth) Authenticated() bool {
-	return s.clientAuth != nil
+	if s.clientAuth != nil {
+		return true
+	} else if existingDeviceCode := s.cfg.GetString(deviceCodeConfigKey); existingDeviceCode != "" && s.bearerToken == "" {
+		// Check if the device is still authenticated with the Platform and if so, get a token.
+		deviceCode := strfmt.UUID(existingDeviceCode)
+		params := oauth.NewAuthDeviceGetParams()
+		params.SetDeviceCode(deviceCode)
+		if response, err := mono.Get().Oauth.AuthDeviceGet(params); response != nil {
+			s.storeAuthenticatedDevice(deviceCode, response)
+		} else {
+			// Either the token for deviceCode has expired, we are rate-limited by the Platform and
+			// have to try again later, or the Platform is unreachable.
+			// Rate limiting can happen during testing.
+			if badRequest, ok := err.(*oauth.AuthDeviceGetBadRequest); ok && *badRequest.Payload.Error == "slow_down" {
```

Comment:
You can use the generated swagger-generated constant rather than hard-coding the string. See: https://github.com/ActiveState/cli/pull/1716/files#diff-f57554ac911491c800c556763e8e626f0cdc6f7fc54d7757ef21e19000634fa3R222